media_remover/src/main/scala/MediaNotRequiredMessageProcessor.scala

```diff
@@ -517,207 +576,427 @@ class MediaNotRequiredMessageProcessor(asLookup: AssetFolderLookup)(
+          logger.warn(s"Something went wrong while trying to get metadata for $nearlineId: $err")
+          Left(s"Something went wrong while trying to get metadata to delete ATT files for $nearlineId: $err")
+      })
 
     combinedRes.map {
-      case (Right(_), Right(_), Right(_)) => Right("Smooth sailing, ATT files removed")
-      case (_, _, _) => Left("One or more ATT files could not be removed. Please look at logs for more info")
+      case (Right(_), Right(_), Right(_)) => Right(s"The ATT files that were present for $nearlineId have been deleted")
+      case (_, _, _) => Left(s"One or more ATT files for $nearlineId could not be deleted. Please look at logs for more info")
+      case Left(err: String) => Left(err)
     }
   }
 
 
-  def deleteMediaFromNearline(vault: Vault, mediaTier: String, filePathMaybe: Option[String], nearlineIdMaybe: Option[String], vidispineItemIdMaybe: Option[String]): Future[Either[String, MessageProcessorReturnValue]] = {
+  def deleteMediaFromNearline(vault: Vault, mediaTier: String, filePathMaybe: Option[String], nearlineIdMaybe: Option[String], vidispineItemIdMaybe: Option[String]): Future[Either[String, MessageProcessorReturnValue]] =
     (mediaTier, filePathMaybe, nearlineIdMaybe) match {
       case ("NEARLINE", Some(filepath), Some(nearlineId)) =>
-        dealWithAttFiles(vault, nearlineId, filepath)
-        // TODO do we need to wrap this with a Future.fromTry?
-        Try { vault.getObject(nearlineId).delete() } match {
-          case Success(_) =>
-            logger.info(s"Nearline media oid=$nearlineId, path=$filepath removed")
-            Future(Right(MediaRemovedMessage(mediaTier = mediaTier, originalFilePath = filepath, nearlineId = Some(nearlineId), vidispineItemId = vidispineItemIdMaybe).asJson))
-          case Failure(exception) =>
-            logger.warn(s"Failed to remove nearline media oid=$nearlineId, path=$filepath, reason: ${exception.getMessage}")
-            Future(Left(s"Failed to remove nearline media oid=$nearlineId, path=$filepath, reason: ${exception.getMessage}"))
-        }
+        // Wait for ATT files to be handled before we delete the main object, since we need the metadata of the main object
+        dealWithAttFiles(vault, nearlineId, filepath).flatMap({
+          case Left(err) =>
+            logger.warn(err)
+            deleteMainNearlineMedia(vault, nearlineId, filepath, vidispineItemIdMaybe)
+          case Right(value) =>
+            logger.debug(value)
+            deleteMainNearlineMedia(vault, nearlineId, filepath, vidispineItemIdMaybe)
+        })
       case (_, _, _) => throw new RuntimeException(s"Cannot delete from nearline, wrong media tier ($mediaTier), or missing nearline id ($nearlineIdMaybe)")
     }
+
+
+  private def deleteMainNearlineMedia(vault: Vault, nearlineId: String, filepath: String, vidispineItemIdMaybe: Option[String]): Future[Either[String, Json]] = {
+    // TODO do we need to wrap this with a Future.fromTry?
```

Comment:
In answer to your question - no, we don't have to. We _could_ do, in order to fomulate the function a bit differently, but I don't think it will actually make improve the readability or aesthetics though.

Reply:
Alright; removing the `TODO`.